code/modules/reagents/chemistry/machinery/chem_mass_spec.dm

```diff
@@ -37,157 +32,241 @@ This will not clean any inverted reagents. Inverted reagents will still be corre
+		. += span_notice("Its Output beaker can be ejected with [EXAMINE_HINT("RMB Alt")] click.")
+	else
+		. += span_warning("Its missing an output beaker, insert with [EXAMINE_HINT("Right Click")].")
+
+	if(anchored)
+		. += span_notice("Its [EXAMINE_HINT("anchored")] in place.")
+	else
+		. += span_warning("Needs to be [EXAMINE_HINT("wrenched")] to use.")
+	. += span_notice("Its maintainence panel can be [EXAMINE_HINT("screwed")] [panel_open ? "closed" : "open"].")
+	if(panel_open)
+		. += span_notice("It can be [EXAMINE_HINT("pried")] apart.")
 
 /obj/machinery/chem_mass_spec/update_overlays()
 	. = ..()
+
 	if(panel_open)
 		. += mutable_appearance(icon, "[base_icon_state]_panel-o")
+		return
 
-/obj/machinery/chem_mass_spec/wrench_act(mob/living/user, obj/item/tool)
+	if(!QDELETED(beaker1))
+		. += "HPLC_beaker1"
+	if(!QDELETED(beaker2))
+		. += "HPLC_beaker2"
+
+	if(is_operational && !panel_open && anchored && !(machine_stat & (BROKEN | NOPOWER)))
+		if(processing_reagents)
+			. += "HPLC_graph_active"
+		else if (length(beaker1?.reagents.reagent_list))
+			. += "HPLC_graph_idle"
+
+/obj/machinery/chem_mass_spec/update_icon_state()
+	if(is_operational && !panel_open && anchored && !(machine_stat & (BROKEN | NOPOWER)))
+		icon_state = "HPLC_on"
+	else
+		icon_state = "HPLC"
+	return ..()
+
+/obj/machinery/chem_mass_spec/Exited(atom/movable/gone, direction)
 	. = ..()
-	default_unfasten_wrench(user, tool)
-	return ITEM_INTERACT_SUCCESS
+	if(gone == beaker1)
+		beaker1 = null
+	if(gone == beaker2)
+		beaker2 = null
 
-/*			beaker swapping/attack code			*/
-/obj/machinery/chem_mass_spec/attackby(obj/item/item, mob/user, params)
-	if(processing_reagents)
-		to_chat(user, "<span class='notice'> The [src] is currently processing a batch!")
+/obj/machinery/chem_mass_spec/RefreshParts()
+	. = ..()
+
+	cms_coefficient = 1
+	for(var/datum/stock_part/micro_laser/laser in component_parts)
+		cms_coefficient /= laser.tier
+
+/obj/machinery/chem_mass_spec/item_interaction(mob/living/user, obj/item/item, list/modifiers, is_right_clicking)
+	if((item.item_flags & ABSTRACT) || (item.flags_1 & HOLOGRAM_1) || !can_interact(user) || !user.can_perform_action(src, FORBID_TELEKINESIS_REACH))
 		return ..()
 
-	if(default_deconstruction_screwdriver(user, icon_state, icon_state, item))
-		update_appearance()
-		return
+	if(processing_reagents)
+		balloon_alert(user, "still processing")
+		return ..()
 
-	if(is_reagent_container(item) && !(item.item_flags & ABSTRACT) && item.is_open_container())
+	if(is_reagent_container(item) && item.is_open_container())
+		. = ITEM_INTERACT_SUCCESS
 		var/obj/item/reagent_containers/beaker = item
-		. = TRUE //no afterattack
 		if(!user.transferItemToLoc(beaker, src))
 			return
-		replace_beaker(user, BEAKER1, beaker)
-		to_chat(user, span_notice("You add [beaker] to [src]."))
+		replace_beaker(user, !is_right_clicking, beaker)
+		to_chat(user, span_notice("You add [beaker] to [is_right_clicking ? "Output" : "Input"] slot."))
```

Comment:
```suggestion
		to_chat(user, span_notice("You add [beaker] to [is_right_clicking ? "output" : "input"] slot."))
```
Though this could be a balloon alert
```suggestion
		balloon_alert(user, "added to [is_right_clicking ? "output" : "input"]")
```
Up to you